tools/sof-dump-status.py

```diff
@@ -87,10 +87,20 @@ def loadPCI(self):
                     break
             pci_info['hw_id']="0x" + tmp_line[2] + tmp_line[1] + " 0x" + tmp_line[4] + tmp_line[3]
             pci_info['hw_name'] = self._pci_ids["0x" + tmp_line[4] + tmp_line[3]]
-            if pci_info['hw_name'] in apci_key_lst:
-                pci_info['device_id'] = self._acpi_ids[pci_info['hw_name']]
             self.pci_lst.append(pci_info)
 
+    def loadACPI(self):
+        self.acpi_lst.clear()
+
+        for mach in self._acpi_ids.keys():
+            acpi_info = {}
+            exit_code, output=subprocess.getstatusoutput("cat /sys/bus/acpi/devices/%s:00/status" % (self._acpi_ids[mach]))
```

Comment:
`getstatusoutput()` has been deprecated for a long time. This shebang requires Python3 anyway.

I see it's already used left  and right but here no `subprocess` seems needed at all to just read a file?! So no need to copy/paste crap, instead:
```python
sys_name = "/sys/bus/acpi/devices/%s:00/status" % self._acpi_ids[mach]
try:
   with open(sys_name) as f:
       output = f.readline().strip()
except FileNotFoundError:
    print(sys_name + ' not found')
    continue
```

If this file should always exist then you can drop the `try: except` entirely and it becomes just 2.5 lines.

This will provide a good template to fix the rest later.

Reply:
I am not going to solve everything in this script. The suggested changes can be done separately.

Reply:
> I am not going to solve everything in this script. The suggested changes can be done separately.

Good I'm not asking to change any existing code.

Reply:
isnt os.path.exists(path) enough?

Reply:
`cat` reads the file.

Reply:
yes, and the missing part is actually to test that the status == 15, which indicates that the device wasn't disabled. Sometimes you see devices but the OS is not supposed to use them for some reason. I didn't add this part but it's needed for consistency